cincinnati/src/lib.rs

```diff
@@ -940,49 +940,63 @@ pub mod testing {
     /// Compares two Graphs and gives a verbose error if not.
     pub fn compare_graphs_verbose(
         mut left: Graph,
+        left_name: &str,
         mut right: Graph,
+        right_name: &str,
         unwanted_metadata_keys: &[&str],
     ) -> Result<(), Box<dyn std::error::Error>> {
         let removed_keys_left = remove_release_metadata(&mut left, unwanted_metadata_keys);
         let removed_keys_right = remove_release_metadata(&mut right, unwanted_metadata_keys);
+        let edges_left = right.get_edges(false)?;
+        let edges_right = left.get_edges(false)?;
+        let releases_left: std::collections::BTreeSet<Release> = (&left).into();
+        let releases_right: std::collections::BTreeSet<Release> = (&right).into();
 
-        if left == right {
+        if edges_left == edges_right && releases_left == releases_right {
```

Comment:
```console
$ git --no-pager log -1
commit b7c4ac5a109f881d662ba6d611115bac458160c0 (HEAD -> graph-data-version, wking/graph-data-version)
Author: W. Trevor King <wking@tremily.us>
Date:   Fri Oct 30 10:15:46 2020 -0700

    cincinnati/src/lib: Ignore node order in graph comparison
    
    Node order has no semantic meaning, but edges do link nodes by index.
    To avoid false-positives like [1]:
    
      $ curl -s https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cincinnati/314/pull-ci-openshift-cincinnati-master-e2e/1322037538348601344/build-log.txt | less
      ...
      ---- e2e_channel_success::_stable_4_2_s390x_ stdout ----
      Querying http://cincinnati-policy-engine-cincinnati-e2e.apps.ci-op-vdj2nvc4-90ac5.origin-ci-int-aws.dev.rhcloud.com/api/upgrades_info/v1/graph?channel=stable-4.2&arch=s390x
      thread 'e2e_channel_success::_stable_4_2_s390x_' panicked at 'Graphs differ! Showing differences from expected to actual', e2e/tests/e2e.rs:68:9
      stack backtrace:
      ...
    
    compare the nodes (after removing unwanted metadata keys) and edges
    (after converting to version-name indexing), instead of comparing the
    graphs directly.
    
    [1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cincinnati/314/pull-ci-openshift-cincinnati-master-e2e/1322037538348601344
```

I dunno why `Graph:eq` wasn't happy with that, but it said the graphs differed and then listed no differences.

Reply:
I would not completely rule out a bug in `Graph::eq` but on suspicion I'd prefer to not close our eyes and walk away from it ;-) Let's see if we can reproduce it. Would you mind extracting the text fixture bump into a separate PR or do you prefer me doing it?

Reply:
Pushed as #343.

Reply:
I've [dropped][1] this change.

[1]: https://github.com/openshift/cincinnati/pull/314#issuecomment-720002421